### PR TITLE
[AF-1555]: Upgrade to Wildfly14

### PR DESCRIPTION
**Pull request by adrielparedes**
## Description

Upgrate to Wildfly 14

There is a concern related to SNAPSHOT version that will be solved in jboss-integration-platform-bom

## Related to:
* https://github.com/jboss-integration/jboss-integration-platform-bom/pull/427
* https://github.com/errai/errai/pull/363
* https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/843
* https://github.com/kiegroup/appformer/pull/483
* https://github.com/kiegroup/kie-wb-common/pull/2147
* https://github.com/kiegroup/jbpm/pull/1336
* https://github.com/kiegroup/jbpm-wb/pull/1206
* https://github.com/kiegroup/drools/pull/2086
* https://github.com/kiegroup/droolsjbpm-integration/pull/1583
* https://github.com/kiegroup/kie-wb-distributions/pull/821 
* https://github.com/kiegroup/kie-docs/pull/1108
* https://github.com/kiegroup/optaplanner/pull/437
* https://github.com/kiegroup/optaplanner-wb/pull/304

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,7 +8,7 @@
     <groupId>org.jboss.integration-platform</groupId>
     <artifactId>jboss-integration-platform-bom</artifactId>
     <!-- Keep in sync with ip-parent version in kie-user-bom-parent/pom.xml -->
-    <version>8.1.0.Final</version>
+    <version>8.2.0-SNAPSHOT</version>
```
Comment (jhrcek) [inline pom.xml]: Probably shouldn't use IP BOM snapshot version 8.2.0-SNAPSHOT

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by mareknovotny** [comment]
What about https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/52a30a5d132f71765157db8b2a6f959aea3f55f5/pom.xml#L244 ? Undertow in Wildfly 14.0.1 is 2.0.13.Final
There is also @mswiderski PR #801 for Wildfly 13 where is effectively upgrade of jackson https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/801/files#diff-600376dffeb79835ede4a0b285078036R84 but I can't see it here

**Comment by mswiderski** [comment]
@mareknovotny undertow is used by smart router but it is completely standalone so I wouldn't make it as part of this upgrade but make sense to upgrade it as well. Maybe separate jira...

**Comment by mareknovotny** [comment]
@mswiderski ok, thanks for quick response.

**Comment by baldimir** [comment]
Any update about this?

**Comment by jhrcek** [comment]
@baldimir the current status is that there is a last bunch of container tests in jbpm that are failing (see PR build of https://github.com/kiegroup/jbpm/pull/1336). @adrielparedes asked @mswiderski and @MarianMacik for help as he's unable to reproduce those failures locally.

**Comment by adrielparedes** [comment]
Jenkins retest this

**Pull request merged**